runtime/fundamentals/modules.md

```diff
@@ -0,0 +1,397 @@
+official standard for JavaScript modules, allows for better tree-shaking,
+improved tooling integration, and native support across different environments.
+
+By adopting ECMAScript modules, Deno ensures compatibility with the evolving
+JavaScript ecosystem, providing developers with a streamlined and predictable
+module system that eliminates the complexities associated with legacy module
+formats like CommonJS.
+
+## Importing modules
+
+In this example the `add` function is imported from a local `calc.ts` module.
+
+```ts title="calc.ts"
+export function add(a: number, b: number): number {
+  return a + b;
+}
+```
+
+```ts title="main.ts"
+// imports the `calc.ts` module next to this file
+import { add } from "./calc.ts";
+
+console.log(add(1, 2)); // 3
+```
+
+You can run this example by calling `deno run main.ts` in the directory that
+contains both `main.ts` and `calc.ts`.
+
+With ECMAScript modules, local import specifiers must always included the full
+file extension. It cannot be omitted.
+
+```ts title="example.ts"
+// WRONG: missing file extension
+import { add } from "./calc";
+
+// CORRECT: includes file extension
+import { add } from "./calc.ts";
+```
+
+## Importing third party modules and libraries
+
+Using third modules in Deno uses the same `import` syntax as your local code.
+Third party modules are typically imported from a remote registry and start with
+`jsr:` , `npm:` or `https://` for URL imports.
+
+```ts title="main.ts"
+import { camelCase } from "jsr:@luca/cases@1.0.0";
+import { say } from "npm:cowsay@1.6.0";
+import { pascalCase } from "https://deno.land/x/case/mod.ts";
+```
+
+You'll find most Deno-related code on [JSR](https://jsr.io), which is the
+recommended registry to use with Deno.
+
+## Managing third party modules and libraries
+
+Typing out the module with the full version specifier can become tedious real
+fast when you're importing them from multiple files. We can centralize the
+management of remote modules via the `imports` key in `deno.json`. The `imports`
+key is used to remap specifiers. We call this the import map.
+
+```json title="deno.json"
+{
+  "imports": {
+    "@luca/cases": "jsr:@luca/cases@^1.0.0",
+    "cowsay": "npm:cowsay@^1.6.0",
+    "cases": "https://deno.land/x/case/mod.ts"
+  }
+}
+```
+
+With the remapped specifiers our code gets a bit simpler:
+
+```ts title="main.ts"
+import { camelCase } from "@luca/cases";
+import { say } from "cowsay";
+import { pascalCase } from "cases";
+```
+
+The remapped name can be any valid specifier. It's a very powerful feature in
+Deno that can remap anything. Learn more about everything the import map can do
+[here](/runtime/manual/basics/import_maps/).
+
+## Adding dependencies with deno add
+
+The installation process is made easier with the `deno add` subcommand. It will
+automatically add the latest version of the package you requested to the
+`imports` section in `deno.json`.
```

Comment:
```suggestion
The installation process is made easy with the `deno add` subcommand. It will
automatically add the latest version of the package you requested to the
`imports` section in `deno.json`.
```